### ISSUE TITLE
Missing Hyper-Parameters tuning module/support

### ISSUE THREAD

**Issue by gbladislau** [oumi-ai/oumi#1984]
I was looking at how to use oumi paired with optuna for a hyperparameter search job.
However, there is nothing regarding this subject in any of oumi's docs, code, etc...
I'm proposing a module, to do an encapsulation of this inside oumi's framework.

I'll start this in my fork of oumi, but every help is well appreciated.

**Comment by gbladislau** [comment]
https://github.com/gbladislau/oumi

**Comment by kaisopos** [comment]
If you find a way to properly do this, please consider contributing it back to Oumi, so other users can take advantage of this!

**Comment by gbladislau** [comment]
My plan is to open a push request later on, so let's see.
